src/lib/Components/Bidding/Components/Input.tsx

```diff
@@ -19,13 +22,17 @@ export class Input extends Component<InputProps, InputState> {
     }
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.setState({
-      borderColor: nextProps.error ? "red100" : "black10",
-    })
+  componentDidUpdate(prevProps) {
+    if (this.props.error !== prevProps.error) {
+      this.setState({ borderColor: this.props.error ? "red100" : "black10" })
+    }
   }
 
   onBlur() {
+    if (this.props.customOnBlur) {
```

Comment:
For our case, we pass in a `validate` function that we want to call `onBlur`. However, passing it in as a prop named `onBlur` meant that _this_ `onBlur` method was never being called. I renamed the one we pass in to `customOnBlur`.